test/module/SmartContractOwnershipRegistry.Module.specs.ts

```diff
@@ -0,0 +1,491 @@
+            randomContract: randomContract
+        }
+    });
+
+    describe("initForSmartAccount(): ", async()=>{
+        it("Calls initForSmartAccount and it successfully initializes userSA with Smart Contract as Owner", async()=>{
+            const {smartContractOwnershipRegistryModule,smartAccountOwnerContract1} = await setupTests();
+
+            let smartContractOwnerhsipSetupData = smartContractOwnershipRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [smartAccountOwnerContract1.address]
+            );
+            const userSA = await getSmartAccountWithModule(smartContractOwnershipRegistryModule.address, smartContractOwnerhsipSetupData,smartAccountDeploymentIndex+1);
+            expect(await smartContractOwnershipRegistryModule.getOwner(userSA.address)).to.be.equal(smartAccountOwnerContract1.address);
+        });
+
+        it("Reverts when setting up EOA as Smart Account Owner", async()=>{
+            const {saFactory,smartContractOwnershipRegistryModule} = await setupTests();
+
+            let smartContractOwnershipSetupData = smartContractOwnershipRegistryModule.interface.encodeFunctionData(
+            "initForSmartAccount",
+            [alice.address]
+            );
+            const expectedSmartAccountAddress = await saFactory.getAddressForCounterFactualAccount(smartContractOwnershipRegistryModule.address,smartContractOwnershipSetupData,smartAccountDeploymentIndex);
+            await expect(saFactory.deployCounterFactualAccount(smartContractOwnershipRegistryModule.address,smartContractOwnershipSetupData,smartAccountDeploymentIndex)).to.be.revertedWith("NotSmartContract");
+            await expect(smartContractOwnershipRegistryModule.getOwner(expectedSmartAccountAddress)).to.be.revertedWith("NoOwnerRegisteredForSmartAccount");
+        });
+
+        it("Reverts when calling again after initialization", async()=>{
+            const {smartContractOwnershipRegistryModule,ecdsaRegistryModule,entryPoint,userSA} = await setupTests();
+            const txnData = ecdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [bob.address],
+            );
+            let userOp = await makeSARegistryModuleUserOp(
+                "executeCall",
+                [smartContractOwnershipRegistryModule.address,0,txnData],
+                 userSA.address,
+                 baseSmartAccountOwner1,
+                 entryPoint,
+                 smartContractOwnershipRegistryModule.address,
+                 ecdsaRegistryModule.address
+            );
+            const tx = await entryPoint.handleOps([userOp], alice.address);
+            await expect(tx).to.emit(entryPoint, "UserOperationRevertReason");
+        });
+    });
+
+    describe("transferOwnership(): ", async()=>{
+        it("Call transferOwnership from userSA and it successfully changes owner", async()=>{
```

Comment:
test name should better describe the test purpose

transferOwnership(): 
         should change owner when called by Smart Account when called from this Smart Account